parsec/scheduling.c

```diff
@@ -185,13 +184,8 @@ int __parsec_execute( parsec_execution_stream_t* es,
             if( PARSEC_HOOK_RETURN_ASYNC != rc ) {
                 /* Let's assume everything goes just fine */
                 task->status = PARSEC_TASK_STATUS_COMPLETE;
-                parsec_device_module_t *dev = NULL;
-                if(PARSEC_DEV_CPU == tc->incarnations[chore_id].type) {
-                    dev = parsec_mca_device_get(0);
-                    parsec_atomic_fetch_inc_int64((int64_t*)&dev->executed_tasks);
-                }
-                else if(PARSEC_DEV_RECURSIVE == tc->incarnations[chore_id].type) {
-                    dev = parsec_mca_device_get(1);
+                if(PARSEC_DEV_RECURSIVE >= tc->incarnations[chore_id].type) {
```

Comment:
We have a macro to find this in a 'portable' way: `parsec_mca_device_is_gpu`
The removal of the counting for the CPUs will break again the report for CPU tasks executed?

Reply:
The incarnations array is packed, I cannot use the `chore_id` for figure out the device.